### DISCUSSION TITLE
Is this the functionality we're trying to realize?

### DISCUSSION THREAD

**Discussion by rmast** [rmast/revealshapes#2] [Q&A]
@jsbien
I found this code:
https://github.com/jsbien/jwilk-marasca-wbl/blob/wbl/misc/augment-djvu-corpus
Is this the functoinality that we're trying to realize?

**Comment by jsbien** [comment]
No. Wilk's tool is just a step in the workflow to create a DjVu corpora, some of them are still available at https://szukajwslownikach.uw.edu.pl/ (and perhaps still used...). If you want to know more, there is a 37 pages documentation in English by Radosław Moszczyński which is available here: https://bitbucket.org/jsbien/poliqarp-for-djvu-documentation-fork.

To make a long story short, I'm interested in "glyph spotting" in old texts and I'm looking for a tool which is more convenient than djvudict and which can serve as a kind of companion for djview4shapes; the latter was implemented in a hurry in the last days of my grant and therefore there are some features missing.

**Comment by rmast** [comment]
I did not find the specification of that djvu enhanced poliqarp index in the repo's I've searched so far. I found some sphinx-build2-documentation that was lacking that specification. If it is in that offline document that would probably be convenient.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jsbien** [comment]
https://en.wikipedia.org/wiki/Poliqarp

It is no longer maintained and used, the site https://szukajwslownikach.uw.edu.pl/ is running without any maintenance... When it breaks nobody will repair it...

I think the only documentation of its indexes was the M.S. thesis (in Polish) of Daniel Janus (https://danieljanus.pl/). The indexes were created from XCES (https://en.wikipedia.org/wiki/XCES) sources.

As for DjVu enhanced indexes, the documentation was never written. In the grant we had limited time and manpower, so some sacrifices were necessary. We have only Wilk's Python scripts and the indexes used on the site.

**Comment by rmast** [comment]
I saw XCES as part of the sphinx-build2-documentation. But the poliqarp-index is not in XML, while XCES talks about XML-format. That XCES-format in that documentation doesn't handle coordinates or sizes.

What part of djview-poliqarp handles these coordinates?

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rmast** [comment]
If I would download the 32 bit Windows installer, would that talk to that unmaintained Polish website? If I visit that website I can enter a query that downloads a .djvu, but I don't know how to view the raw index itself.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jsbien** [reply]
The raw index is not accessible to users. If you want to see a sample, I can provide you with it. I don't think however that it is relevant in any way to revealshapes (or vice versa). Am I missing something?

> If I would download the 32 bit Windows installer

Installer of what?

Just checked on Linux that djview4poliqarp still talks to the site, the Windows version should also do. 
![Screenshot from 2024-04-15 22-57-05](https://github.com/rmast/revealshapes/assets/8024702/cfb739e6-11d6-4731-bd8d-bf2b5bdf22a5)

**Comment by jsbien** [comment]
> What part of djview-poliqarp handles these coordinates?

djview4poliqarp is just a client to the server simulating just a HTML browser. So djview4poliqarp doesn't handle directly any indexes.

On the other hand poliqarp in a narrow sense is just a very limited server, what you see at https://szukajwslownikach.uw.edu.pl/ is just the output of a HTML client to poliqarp named `marasca` by Wilk. So, more precisely, djview4poliqarp is just a client to a client overcoming in particular some limitations of the so called graphical concordances of marasca. The support of *.csv indexes was added later, primarily to handle indexes to Linde's dictionary (https://github.com/jsbien/iLindeCSV).

>  the poliqarp-index is not in XML

We don't use the term "poliqarp-index". I assumed you mean the binary format of the index created from the XCES sources. Anyway it has no relation to the djview4poliqarp *.csv indexes.

**Comment by rmast** [comment]
It does. And when I press h the selected region disappears. So that client-application must know those index-lines with coordinates of the search results.

It even contains "index"-menu-items to save and add things. Did not get it save a thing though, but probably when something is selected. I'm not sure the original index won't be altered by those actions, so a little hesitant to choose those.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by jsbien** [reply]
But this is a different index stored locally as a *.csv file. Using it does not involve the server in any way, but you can open a remote DjVu document (which may happen to reside on a Poliqarp server or any other site).

When you save or modify entries you work on a local file.

Actually the name `djview4poliqarp` is now misleading, I considered renaming it to `djview4index` but decided it's too dificult for me with my almost null knowledge of QT/C++.
